tests/handlers/protocols-query.spec.ts

```diff
@@ -293,6 +293,91 @@ export function testProtocolsQueryHandler(): void {
+
+
+          // Alice grants Bob the access to ProtocolsQuery on her DWN for a specific protocol
+          const permissionGrant = await PermissionsProtocol.createGrant({
+            signer      : Jws.createSigner(alice),
+            grantedTo   : bob.did,
+            dateExpires : Time.createOffsetTimestamp({ seconds: 60 * 60 * 24 }),
+            scope       : { interface: DwnInterfaceName.Protocols, method: DwnMethodName.Query, protocol: allowedProtocol }
+          });
+
+          const dataStream = DataStream.fromBytes(permissionGrant.permissionGrantBytes);
+          const grantRecordsWriteReply = await dwn.processMessage(alice.did, permissionGrant.recordsWrite.message, { dataStream });
+          expect(grantRecordsWriteReply.status.code).to.equal(202);
+
+          // Bob tries to ProtocolsQuery to Alice's DWN for the allowed protocol
+          const protocolsQueryAllowed = await TestDataGenerator.generateProtocolsQuery({
+            author : bob,
+            filter : {
+              protocol: allowedProtocol
+            },
+            permissionGrantId: permissionGrant.recordsWrite.message.recordId
+          });
+
+          const protocolQueryAllowedReply = await dwn.processMessage(alice.did, protocolsQueryAllowed.message);
+          expect(protocolQueryAllowedReply.status.code).to.equal(200);
+          expect(protocolQueryAllowedReply.entries?.length).to.equal(1);
+          expect(protocolQueryAllowedReply.entries![0].descriptor.definition.protocol).to.deep.equal(allowedProtocol);
+
+          // Bob tries to ProtocolsQuery to Alice's DWN for a different protocol
+          const protocolQueryNotAllowed = await TestDataGenerator.generateProtocolsQuery({
+            author : bob,
+            filter : {
+              protocol: notAllowedProtocol,
+            },
+            permissionGrantId: permissionGrant.recordsWrite.message.recordId
+          });
+
+          const protocolQueryNotAllowedReply = await dwn.processMessage(alice.did, protocolQueryNotAllowed.message);
+          expect(protocolQueryNotAllowedReply.status.code).to.equal(200);
+          expect(protocolQueryNotAllowedReply.entries?.length).to.equal(0);
+
+          // Bob tries to ProtocolsQuery to Alice's DWN for a published protocol with the same grant
+          const protocolQueryPublished = await TestDataGenerator.generateProtocolsQuery({
+            author : bob,
+            filter : {
+              protocol: publishedProtocol,
+            },
+            permissionGrantId: permissionGrant.recordsWrite.message.recordId
+          });
+
+          const protocolQueryPublishedReply = await dwn.processMessage(alice.did, protocolQueryPublished.message);
+          expect(protocolQueryPublishedReply.status.code).to.equal(200);
+          expect(protocolQueryPublishedReply.entries?.length).to.equal(1);
+          expect(protocolQueryPublishedReply.entries![0].descriptor.definition.protocol).to.deep.equal(publishedProtocol);
+        });
```

Comment:
Really curious to know what would happen if grant is invoked on a query without any filter on protocol?

Reply:
Added a test case to show that for future readers.

Reply:
I was more referring to a grant that is able to query all protocols, and no protocol filter is specified in a query. Is this possible? What should be the expectation? All private and public protocols should be returned? Do we have a test for this?

Reply:
Ah I get what you're saying.

Good call out, I added it to the `allows an external party to ProtocolsQuery only if they have a valid grant` test above.